### PR TITLE
fixing available_bucket, which was causing errors in trade etc

### PR DESCRIPTION
**Pull request by infobiac**

### CODE REVIEW

File: trading/models.py [hunk 1]
```diff
@@ -38,15 +38,17 @@ def available_cash(self):
...
         return stock_val + bucket_val
 
-    @staticmethod
-    def available_buckets(bkt):
+    def available_buckets(self, bkt):
         """
         Find the available buckets that have quantity > 0
         """
-        bucket = TradeBucket.buckettrades.filter(bucket=bkt).annotate(
+        bucket = self.buckettrades.all().filter(stock=bkt.id).annotate(
```
Comment (Neitsch) [inline trading/models.py]: First filter then all

File: trading/models.py [hunk 2]
```diff
@@ -38,15 +38,17 @@ def available_cash(self):
...
             sum_quantity=models.Sum('quantity'))
```
Comment (Neitsch) [inline trading/models.py]: Use get instead of all, since we can have at most one after aggregation

Comment (Neitsch) [reply trading/models.py]: Wrap the get in a try catch (Python EAFP)

File: trading/models.py [hunk 3]
```diff
@@ -38,15 +38,17 @@ def available_cash(self):
...
-        if bucket:
-            return bucket[0].sum_quantity
+        total_sum = 0
+        for x in bucket:
+            total_sum += x.sum_quantity
+        if total_sum is not 0:
+            return total_sum
```
Comment (Neitsch) [inline trading/models.py]: These two lines are not necessary, but you should not need these 5 lines after you addresse the comments

**Review by Neitsch — CHANGES_REQUESTED** [review]

File: trading/models.py (hunk 2 above)
Comment (infobiac) [reply trading/models.py]: But doesn't the all have to evaluate before the aggregation, so there's something to aggregate on?

File: trading/models.py (hunk 3 above)
Comment (infobiac) [reply trading/models.py]: unfortunately the filter/sum combo doesn't actually compute the sum across all of them... so this step is necessary :(

File: trading/models.py [hunk 4]
```diff
@@ -38,26 +38,27 @@ def available_cash(self):
...
-        bucket = TradeBucket.buckettrades.filter(bucket=bkt).annotate(
+        buckets = self.buckettrades.filter(stock=bkt.id).all().annotate(
             sum_quantity=models.Sum('quantity'))
-        if bucket:
-            return bucket[0].sum_quantity
-        return 0
+        total_sum = 0
+        for bucket in buckets:
+            total_sum += bucket.sum_quantity
+        return total_sum
```
Comment (Neitsch) [inline trading/models.py]: I think you can do better:
```
try:
    return self.buckettrades.filter(stock=bkt).annotate(sum_quantity=models.Sum('quantity')).get().sum_quantity
except ObjectDoesNotExist:
    return 0

File: trading/models.py [hunk 5]
```diff
@@ -38,26 +38,27 @@ def available_cash(self):
...
-        stock = TradeStock.trades.filter(stock=stk).annotate(sum_quantity=models.Sum('quantity'))
-        if stock:
-            return stock[0].sum_quantity
-        return 0
+        stocks = self.trades.filter(stock=stk).all().annotate(
+            sum_quantity=models.Sum('quantity'))
+        total_sum = 0
+        for stock in stocks:
+            total_sum += stock.sum_quantity
+        return total_sum
```
Comment (Neitsch) [inline trading/models.py]: Same here
```
try:
    return self.trades.filter(stock=bkt).annotate(sum_quantity=models.Sum('quantity')).get().sum_quantity
except ObjectDoesNotExist:
    return 0
```

**Review by Neitsch — CHANGES_REQUESTED** [review]

**Review by Neitsch — APPROVED** [review]
Feel free to merge, I'll look into a more efficient way

**Pull request merged**